tests/appsec/waf/test_telemetry.py

```diff
@@ -1,28 +1,224 @@
+        # XXX: hack to prevent rid inhibiting the dd-test-scanner-log-block rule
+        rid_in_user_agent=False,
+    )
+    Test_TelemetryMetrics.__common_setup_done = True
+
+
+@rfc("https://docs.google.com/document/d/1qBDsS_ZKeov226CPx2DneolxaARd66hUJJ5Lh9wjhlE")
+@released(python="?", cpp="?", golang="?", java="1.12.0", dotnet="?", nodejs="?", php="?", ruby="?")
+@scenarios.appsec_waf_telemetry
+class Test_TelemetryMetrics:
+    """Test instrumentation telemetry metrics, type of metrics generate-metrics"""
+
+    setup_all_telemetry_requests_are_successful = _setup
+
+    def test_all_telemetry_requests_are_successful(self):
+        """Tests that all telemetry requests succeed."""
+        for data in interfaces.library.get_telemetry_data():
+            assert data["response"]["status_code"] == 202
+
+    setup_headers_are_correct = _setup
+
+    @bug(context.library < "java@1.13.0", reason="Missing two headers")
+    @missing_feature(context.weblog_variant == "spring-boot-native", reason="GraalVM. Tracing support only")
+    @missing_feature(context.weblog_variant == "spring-boot-3-native", reason="GraalVM. Tracing support only")
+    def test_headers_are_correct(self):
+        """Tests that all telemetry requests have correct headers."""
+        for data in interfaces.library.get_telemetry_data():
+            request_type = data["request"]["content"].get("request_type")
+            _validate_headers(data["request"]["headers"], request_type)
+
+    setup_metric_waf_init = _setup
+
+    @missing_feature(context.weblog_variant == "spring-boot-native", reason="GraalVM. Tracing support only")
+    @missing_feature(context.weblog_variant == "spring-boot-3-native", reason="GraalVM. Tracing support only")
+    def test_metric_waf_init(self):
+        """Test waf.init metric."""
+        expected_metric_name = "waf.init"
+        mandatory_tag_prefixes = {
+            "waf_version",
+            "event_rules_version",
+        }
+        valid_tag_prefixes = mandatory_tag_prefixes
+        series = self._find_series(TELEMETRY_REQUEST_TYPE_GENERATE_METRICS, "appsec", expected_metric_name)
+        # TODO(Python). Gunicorn creates 2 process (main gunicorn process + X child workers). It generates two init
+        if context.library == "python":
+            assert len(series) == 2
+        else:
+            assert len(series) == 1
+        s = series[0]
+        assert s["_computed_namespace"] == "appsec"
+        assert s["metric"] == expected_metric_name
+        assert s["common"] is True
+        assert s["type"] == "count"
+
+        full_tags = set(s["tags"])
+        self._assert_valid_tags(
+            full_tags=full_tags, valid_prefixes=valid_tag_prefixes, mandatory_prefixes=mandatory_tag_prefixes
+        )
+
+        assert len(s["points"]) == 1
+        p = s["points"][0]
+        assert p[1] == 1
+
+    setup_metric_waf_updates = _setup
+
+    @irrelevant(reason="Test not implemented")
```

Comment:
you really want to skip this tests for everybody ?

Reply:
At the moment yes, since it requires some additional setup which is not present in the scenario yet. Once we add it, we should replace this with `@missing_feature` or `@released`.

Reply:
Can you use `@missing_feature`? it's for both missing stuff in test or in component.

`@irrelevant` is reserved for test that will never (or very unlikely) be activated.